src/bio.c

```diff
@@ -252,7 +252,12 @@ int wolfSSL_BIO_read(WOLFSSL_BIO* bio, void* buf, int len)
     #endif
 
         if (bio->type == WOLFSSL_BIO_SOCKET) {
+    #ifdef USE_WOLFSSL_IO
             ret = wolfIO_Recv(bio->num, (char*)buf, len, 0);
+    #else
+            WOLFSSL_MSG("Error USE_WOLFSSL_IO not defined");
```

Comment:
How is this fixing IO?

We should we work correctly with default I/O callbacks, with user I/O callback, and with and without wolfSSL IO.

Erroring out at runtime instead of working correctly is better?  Explain why this is best approach for all users in all cases.

Reply:
This was an attempt to fix the build error for users with this config, but not necessarily using BIO sockets. It would be better to fully implement the feature for all use cases.